src/wallet/bip44/bip44.ts

```diff
@@ -144,4 +149,55 @@ export class BIP44 {
             `Failed to derive a valid address for the zone ${zone} after ${MAX_ADDRESS_DERIVATION_ATTEMPTS} attempts.`,
         );
     }
+
+    /**
+     * Performs non-hardened child derivation from a public key and chain code. This allows deriving child addresses at
+     * CPU speed without private key access.
+     *
+     * @param {string} publicKey - The compressed public key (33 bytes as hex string)
+     * @param {string} chainCode - The chain code (32 bytes as hex string)
+     * @param {number} account - The account index (already hardened, e.g., 0' becomes 0)
+     * @param {number} change - The change index (0 for external, 1 for internal)
+     * @param {number} addressIndex - The address index
+     * @returns {Object} Object containing derived public key, chain code, and address
+     */
+    public static deriveChildFromPublic(
+        publicKey: string,
+        chainCode: string,
+        account: number = 0,
+        change: number = 0,
+        addressIndex: number = 0,
+    ): { publicKey: string; chainCode: string; address: string; path: string } {
+        // Create BIP32 factory with ecc
+        const bip32 = BIP32Factory(ecc);
+
```

Comment:
maybe add validation on the index range
if (change >= 0x80000000 || addressIndex >= 0x80000000) {
    throw new Error('Change and address indices must be non-hardened (< 2^31)');
}
something like this

Reply:
Fixed